### PR TITLE
Add an opt-in flag to normalize the output of get_param_schema()

### PR DESCRIPTION
**Pull request by Gallaecio**

### CODE REVIEW

File: scrapy_spider_metadata/_params.py [hunk 1]
```diff
@@ -22,11 +22,14 @@ def __init__(self, *args, **kwargs) -> None:
...
     @classmethod
-    def get_param_schema(cls) -> Dict[Any, Any]:
+    def get_param_schema(cls, normalize=False) -> Dict[Any, Any]:
         """Return a :class:`dict` with the :ref:`parameter definition
         <define-params>` as `JSON Schema`_.
 
         .. _JSON Schema: https://json-schema.org/
+
+        If *normalize* is ``True``, the returned schema will be the same
+        regardless of whether you are using Pydantic 1.x or Pydantic 2.x.
```
Comment (kmike) [inline scrapy_spider_metadata/_params.py:32]: For the context, is it following pydantic 2.x output, pydantic 1.x output, or is it a different format?

**Review by kmike — APPROVED** [review]

File: scrapy_spider_metadata/_params.py (hunk 1 above)
Comment (Gallaecio) [reply scrapy_spider_metadata/_params.py:32]: A different one. From Pydantic 1.x it takes:
- Not using `$defs`.
- Not using the enum class name as a default field `title`, and instead use the field name to generate such a default `title`.

I would mention the differences, but we cannot tell if future versions of Pydantic may remove some of them. Should we explicitly state that the output may not match either?

Comment (wRAR) [reply scrapy_spider_metadata/_params.py:32]: > Should we explicitly state that the output may not match either?

I think so.

File: scrapy_spider_metadata/_params.py [hunk 2]
```diff
@@ -22,11 +22,15 @@ def __init__(self, *args, **kwargs) -> None:
...
-    def get_param_schema(cls) -> Dict[Any, Any]:
+    def get_param_schema(cls, normalize=False) -> Dict[Any, Any]:
         """Return a :class:`dict` with the :ref:`parameter definition
         <define-params>` as `JSON Schema`_.
 
         .. _JSON Schema: https://json-schema.org/
+
+        If *normalize* is ``True``, the returned schema will be the same
+        regardless of whether you are using Pydantic 1.x or Pydantic 2.x. The
+        normalized schema may not match the output of any Pydantic version.
```
Comment (kmike) [inline scrapy_spider_metadata/_params.py:33]: Should we state something like "while still implementing the same validation rules" or something like this?

**Review by wRAR — APPROVED** [review]

File: scrapy_spider_metadata/_params.py (hunk 2 above)
Comment (Gallaecio) [reply scrapy_spider_metadata/_params.py]: https://github.com/scrapy-plugins/scrapy-spider-metadata/pull/4/commits/46dc48c5b199d7b2212e18aaa90ff0ac4a9cdaff ?

File: scrapy_spider_metadata/_params.py [hunk 3]
```diff
@@ -22,11 +22,16 @@ def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
     @classmethod
-    def get_param_schema(cls) -> Dict[Any, Any]:
+    def get_param_schema(cls, normalize=False) -> Dict[Any, Any]:
```
Comment (wRAR) [inline scrapy_spider_metadata/_params.py:25]:
```suggestion
    def get_param_schema(cls, normalize: bool = False) -> Dict[Any, Any]:
```

**Pull request merged**